### DISCUSSION TITLE
What is the recommended way to manage product types and applicable attributes only

### DISCUSSION THREAD

**Discussion by SharadKumar** [swellstores/community#65] [Help] [answered]
Lets say the store sells various types of products, `t-shirts` and `mobile-phones`. 
- **t-shirts** have `size` (S,M,L,XL) and `color` (Red, Blue, Green) for options/variants, and few other attributes.
- **mobile-phones** have totally different aspects such as `brand` (Samsung, Apple), `os` (iOS, Android), and model options (iPhone 13, iPhone 14) etc. These again can be managed through combination of attributes, options, and variants, as already documented and available on the dashboard.

However, what is not clear is how to address during onboarding and setup implicitly such that the above two (for example) product types allow for, either:
- Dashboard to display relevant attributes/options based on the type: `t-shirt` vs `mobile-phone`
- Category could be used (a Category per product type) but they are hierarchical (for good reason) and one product can belong to multiple categories. So clearly they aren't meant for just that, and are envisaged for flexibility i.e. as suited to the store.
- Create a model for each product type based on `product`? How to do this. I reckon we need a proper example described for new starters.

This documentation stays silent on recommended approach to product types:
https://www.swell.is/help/products/products-overview

All in all, please advise what others are doing and what would you recommend?

**Comment by logeshswell** [comment, marked as answer]
@SharadKumar I have a small suggestion for you.
Can we have a custom field `category` in the Attributes model?
This could help you to fetch the list of attributes for a product type.

How to add a new field to a standard model (eg: Attributes)?
Refer to this guide: https://developers.swell.is/guides/data-model-customization#adding-fields-to-a-standard-model
